### DISCUSSION TITLE
Build package with local crates

### DISCUSSION THREAD

**Discussion by nezsha** [ipetkov/crane#651] [Q&A] [answered]
I've been recently building a project that I have a few local crates for that I did not publish and don't intend to.
I wanted to compile one of my the crates but it depends on another local crate and nix fails to find it during the build process.
```nix
 packages = {
        editor = craneLib.buildPackage {
          src = ./editor;
          name = "editor";
          version = "1.1.1";
          nativeBuildInputs = with pkgs; [
            openssl
            pkg-config
            libiconv
          ];
          buildInputs = with pkgs; [
            openssl
            pkg-config
            libiconv
            ffmpeg
          ];

          postInstall = ''
            install -Dt $out/bin editor
            mkdir $out/bin/.wrapped
            mv $out/bin/editor $out/bin/.wrapped/
            makeWrapper "$out/bin/.wrapped/editor "$out/bin/editor \
            --prefix LD_LIBRARY_PATH : ${pkgs.ffmpeg} \
            --prefix PATH : $out/bin
          '';
        };
      };
```

```
     >   failed to load source for dependency `bucket`
       >
       > Caused by:
       >   Unable to update /build/modules/bucket
       >
       > Caused by:
       >   failed to read `/build/modules/bucket/Cargo.toml`
       >
       > Caused by:
       >   No such file or directory (os error 2)
  ```
  The program I want to compile depends on like 2 more local crates, is there a way to compile the project so it gets included into the derivation? The issues I saw about this didn't really help me

**Comment by nezsha** [comment]
I managed to use workspaces with cargo-hakari and now it seems to compile just fine.

**Reply by dpc** [reply]
You shouldn't need to use hakari for it. From my experience if the crate specifies `path` it should work just fine.

**Comment by ipetkov** [comment, marked as answer]
> src = ./editor;

When using `src = ./editor;` this instructs Nix to only make the `./editor` directory available for the derivation. Assuming other workspace crates are in different directories, you will need to change the `src` attribute to include them.

The `quick-start-workspace` example shows how this can be achieved with `lib.fileset`: https://github.com/ipetkov/crane/blob/9791c77eb7e98b8d8ac5b0305d47282f994411ca/examples/quick-start-workspace/flake.nix#L74-L80
